### ISSUE TITLE
Automatically generate avatars based on initials

### ISSUE THREAD

**Issue by kolaente** [go-gitea/gitea#6400]
From the discussion on discourd, it would be nice if there was a way to let Gitea generate avatars for profiles based on the label like this one:
![image](https://user-images.githubusercontent.com/13721712/54718061-fa98c800-4b59-11e9-98ee-c23b97d70307.png)

Possible ways to solve this:

https://stackoverflow.com/questions/38299930/how-to-add-a-simple-text-label-to-an-image-in-go
https://github.com/holys/initials-avatar

cc @mrsdizzie

**Comment by techknowlogick** [comment]
If we are thinking of adding new options to create different avatars I've actually ported a few avatar generation libs to golang that could also be used as well (perhaps in config there would be a setting to choose from a list of libraries for default avatars):

- src.techknowlogick.com/monster-id
- src.techknowlogick.com/wavatars
- src.techknowlogick.com/retricon
- src.techknowlogick.com/mysteryperson-id

**Comment by mrsdizzie** [comment]
An initials based one would be a good/better default for now if you have a preferred existing library to use. I think adding a more solid default first and then focus on providing alternative options would be good and happy to help with that if necessary

**Comment by kolaente** [comment]
@mrsdizzie yes, but maybe it's a good idea to implement it in a way so we can extend it easily afterwards.

**Comment by 0x5c** [comment]
As a user, that does happen to prefer the current behaviour, this definitely needs to be configurable. I can definitely see why one would prefer avatars based on initials though.
